core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMapFactory.java

```diff
@@ -136,9 +136,22 @@ public DataMapBuilder createBuilder(Segment segment, String shardName,
           getTableBlockIndexUniqueIdentifiers(segment);
 
       for (TableBlockIndexUniqueIdentifier tableBlockIndexUniqueIdentifier : identifiers) {
-        tableBlockIndexUniqueIdentifierWrappers.add(
-            new TableBlockIndexUniqueIdentifierWrapper(tableBlockIndexUniqueIdentifier,
-                this.getCarbonTable()));
+        if (null != partitionsToPrune && !partitionsToPrune.isEmpty()) {
+          // add only tableBlockUniqueIdentifier that matches the partition
+          for (PartitionSpec partitionSpec : partitionsToPrune) {
+            if (partitionSpec.getLocation().toString()
+                .equalsIgnoreCase(tableBlockIndexUniqueIdentifier.getIndexFilePath())) {
```

Comment:
It is not easy to understand why comparing partition spec location with index file path, can you make it more readable

Reply:
Actually, `tableBlockIndexUniqueIdentifier.getIndexFilePath()` is the index file parent path. I will add comments for the same